app/containers/message/Video.tsx

```diff
@@ -40,30 +59,147 @@ interface IMessageVideo {
 	isReply?: boolean;
 }
 
+const CancelIndicator = () => {
+	const { colors } = useTheme();
+	return (
+		<View style={styles.cancelContainer}>
+			<Text style={[styles.text, { color: colors.auxiliaryText }]}>{I18n.t('Cancel')}</Text>
+		</View>
+	);
+};
+
+const Thumbnail = ({ loading, video, cached }: { loading: boolean; video: string; cached: boolean }) => {
+	const [thumbnailImage, setThumbnailImage] = useState('');
+
+	useEffect(() => {
```

Comment:
This is running all the time and it's very expensive. You're never caching it.
If it's so costly that we can't make it better, we need to remove it and do it on backend as we discussed.